### PR TITLE
Add swipe right to close SeeRequest

### PR DESCRIPTION
**Pull request by asgalecki**
https://user-images.githubusercontent.com/70109479/178460482-56cc84d7-3834-4258-bd7a-ae36d4547be0.mp4

https://user-images.githubusercontent.com/70109479/178460501-fd8ac2b5-35d9-4f46-890d-a0100cc550f3.mp4

https://user-images.githubusercontent.com/70109479/178460523-973b90a4-ed42-4958-a15c-9aab98f35c4c.mp4

https://user-images.githubusercontent.com/70109479/178464260-060bdf1a-1672-476b-a5da-53fd4c50dd6f.mp4

### CODE REVIEW

File: src/navigation/RequestsNavigation.tsx [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const RequestsNavigation = () => (
     <RequestsStack.Screen
       name="SEE_REQUEST"
       component={SeeRequest}
-      options={{ ...TransitionPresets.ModalSlideFromBottomIOS }}
+      options={{ ...TransitionPresets.SlideFromRightIOS }}
```
Comment (lukbar48) [inline src/navigation/RequestsNavigation.tsx]: please delete redundant spread and braces, and apply GestureRecognizer to handle gestures in SeeRequest screen

Comment (asgalecki) [reply src/navigation/RequestsNavigation.tsx]: I'm waiting for a feedback from design team as I'm not sure do they want solution like this.

File: src/components/RequestDetails/SeeRequest.tsx [hunk 2]
```diff
@@ -45,9 +46,11 @@ export const SeeRequest = ({ route: { params: p } }: RequestsNavigationProps<'SE
         </Text>
         <Box paddingRight="xl" />
       </ModalHeader>
-      <Box marginTop="l" paddingBottom="m" flex={1}>
-        <RequestDetails {...p} showStatus wasSent />
-      </Box>
+      <GestureRecognizer onSwipeRight={goBack} onEnded androidOnly>
+        <Box marginTop="l" paddingBottom="m" flex={1}>
```
Comment (lukbar48) [inline src/components/RequestDetails/SeeRequest.tsx]: you can add style prop to gesture recognizer so we won't have to add extra box here

File: src/utils/GestureRecognizer.tsx [hunk 3]
```diff
@@ -10,6 +11,8 @@ type GestureRecognizerProps = {
   onSwipeUp?: F0
   onSwipeDown?: F0
   onFailed?: F0
+  onEnded?: boolean
+  androidOnly?: boolean
```
Comment (lukbar48) [inline src/utils/GestureRecognizer.tsx]: both props can be  ?: true

File: src/utils/GestureRecognizer.tsx [hunk 4]
```diff
@@ -39,8 +44,16 @@ export const GestureRecognizer = ({
     if (onSwipeUp && touchStartY.current - touchMoveY > minSwipeDistance) onSwipeUp()
   }
 
+  if (androidOnly && isIos) {
+    return <Box style={{ flex: 1 }}>{children}</Box>
```
Comment (lukbar48) [inline src/utils/GestureRecognizer.tsx]: you can replace Box with View as it's already imported

**Review by lukbar48 — APPROVED** [review]

**Pull request merged**